File: AplicacionMusica/app/src/main/java/com/example/aplicacionmusica/modelo/Usuario.kt
package com.example.aplicacionmusica.modelo

data class Usuario(
    val nombre: String = "",
    val apellidos: String = "",
    val correo: String = "",
    val contra: String = "",
    val instrumentos: List<Instrumento>
) {

    val _id: String = ""


    constructor( instrumentos: List<Instrumento>) : this("", "", "", "", instrumentos)

}

File: API-MUSICSHOP/models/usuario.ts
import mongoose, { Schema, Document } from "mongoose";
import IUsuario from "../interfaces/usuario";

const SchemaUsuario: Schema = new Schema<IUsuario>(
    {
      nombre: { type: Schema.Types.String, required: true },
      apellidos: { type: Schema.Types.String, required: true },
      correo: { type: Schema.Types.String, required: true },
      contra: { type: Schema.Types.String, required: true },
      instrumentos: {
        type: [Schema.Types.ObjectId],
        ref: "instrumentos",
      },
    },
    { timestamps: false, versionKey: false }
  );
  
  export const UsuarioModel = mongoose.model<IUsuario & Document>(
    "usuarios",
    SchemaUsuario
  );